packages/aws-rfdk/lib/deadline/scripts/bash/configureWorker.sh

```diff
@@ -36,6 +36,21 @@ if [ ! -f "$DEADLINE_COMMAND" ]; then
     exit 1
 fi
 
+isVersionLessThan() {
+    python -c "import sys;sys.exit(0 if tuple(map(int, sys.argv[-2].split('.'))) < tuple(map(int, sys.argv[-1].split('.'))) else 1)" "$1" "$2"
+}
+
+DEADLINE_VERSION=$("$DEADLINE_COMMAND" -Version | grep -oP '[v]\K\d+\.\d+\.\d+\.\d+\b')
+if [ -z "$DEADLINE_VERSION" ]; then
+    echo "ERROR: Unable to identify the version of installed Deadline Client. Exiting..."
+    exit 1
+fi
+MINIMUM_SUPPORTED_DEADLINE_VERSION="10.1.9"
+if $(isVersionLessThan $DEADLINE_VERSION $MINIMUM_SUPPORTED_DEADLINE_VERSION); then
```

Comment:
I believe [command substitution](https://www.gnu.org/software/bash/manual/html_node/Command-Substitution.html#Command-Substitution) or output expansion (`$(...)`) is not what we want here. Instead we want to check the exit code with:

```suggestion
if isVersionLessThan $DEADLINE_VERSION $MINIMUM_SUPPORTED_DEADLINE_VERSION; then
```

From [conditional constructs](https://www.gnu.org/software/bash/manual/html_node/Conditional-Constructs.html#Conditional-Constructs):

>   The syntax of the if command is:
>
>       if test-commands; then
>           consequent-commands;
>       [elif more-test-commands; then
>           more-consequents;]
>       [else alternate-consequents;]
>       fi
>
>   The _test-commands_ list is executed, and if its return status is zero, the _consequent-commands_ list is executed. If _test-commands_ returns a non-zero status, each `elif` list is executed in turn, and if its exit status is zero, the corresponding _more-consequents_ is executed and the command completes. If ‘`else alternate-consequents`’ is present, and the final command in the final `if` or `elif` clause has a non-zero exit status, then _alternate-consequents_ is executed. The return status is the exit status of the last command executed, or zero if no condition tested true.

Reply:
fixed.

Reply:
Thanks. resolving